src/app/InteractionModelEngine.cpp

```diff
@@ -1853,26 +1853,22 @@ void InteractionModelEngine::ResumeSubscriptionsTimerCallback(System::Layer * ap
             continue;
         }
 
-        auto requestedAttributePathCount = subscriptionInfo.mAttributePaths.AllocatedSize();
-        auto requestedEventPathCount     = subscriptionInfo.mEventPaths.AllocatedSize();
-        if (!imEngine->EnsureResourceForSubscription(subscriptionInfo.mFabricIndex, requestedAttributePathCount,
-                                                     requestedEventPathCount))
+        auto subscriptionResumptionSessionEstablisher = Platform::New<SubscriptionResumptionSessionEstablisher>();
```

Comment:
can we use MakeUnique to handle memory free up for us? then use `release` if you need to release the ownership.

without that, currently the lifetime of this establisher is unclear to me. It goes out of scope of this function allocated ... what deallocates it?

Reply:
Done.